### DISCUSSION TITLE
Disable "pg_autoscaler" failed.

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#9950] [Q&A]
![image](https://user-images.githubusercontent.com/13997135/160355203-c43b8806-d4cc-4f8f-b55f-c2d1c737b221.png)

**Comment by BlaineEXE** [comment]
What version of Ceph is this?

**Comment by zhucan** [comment]
16.2.7

**Comment by travisn** [comment]
@zhucan Please include repro steps. Did you set in the cluster.yaml the pg_autoscaler to `disabled: false`? Or what did you set? And what version of rook?

**Comment by zhucan** [comment]
Only set in the cluster.yaml the pg_autoscaler to disabled: false. and kubectl apply -f cluter.yaml. the version of rook is 1.8.5. @travisn

**Comment by travisn** [comment]
@zhucan I was able to repro this as well. When running the disable command in the toolbox, there are some more details:
```
[rook@rook-ceph-tools-d6d7c985c-lqpl6 /]$ ceph mgr module disable pg_autoscaler   
Error EINVAL: module 'pg_autoscaler' cannot be disabled (always-on)
```
So it appears ceph does not anymore allow disabling the autoscaler.

**Comment by zhucan** [comment]
@travisn Thanks

**Comment by zhucan** [comment]
@travisn Is possible support to create pool with specail pg and configurable model of pg autoscale?

**Reply by zhucan** [reply]
ceph osd pool create <pool> [<pg_num:int>] [<pgp_num:int>] [replicated|erasure] [<erasure_    create pool
 code_profile>] [<rule>] [<expected_num_objects:int>] [<size:int>] [<pg_num_min:int>]    
 [<pg_num_max:int>] [on|off|warn] [--bulk] [<target_size_bytes:int>] [<target_size_      
 ratio:float>]

**Reply by travisn** [reply]
Yes, looks like the PGs can be controlled per-pool as seen in the [Ceph PG docs](https://docs.ceph.com/en/latest/rados/operations/placement-groups/#autoscaling-placement-groups). 

The autoscaler is generally considered more helpful than negative, but if someone needs granular control then support can certainly be added to the pool CR.

**Reply by zhucan** [reply]
If we want to set to pool of rgw，we need to discuss it more deep

**Reply by travisn** [reply]
The pools in the object store use the same spec as the CephBlockPool, so if we make a change to the CephBlockPool, the pools in the CephObjectStore will inherit those settings.

**Reply by zhucan** [reply]
so we need to create a issue to releated it?

**Reply by travisn** [reply]
yes sounds good to create an issue

**Comment by zhucan** [comment]
There are some potential negative effects to be aware of when using Ceph PG autoscaling:

1. Increased latency: Autoscaling can result in increased latency if it takes time for Ceph to rebalance the data across the cluster.
2. Increased network traffic: Autoscaling can result in increased network traffic as data is rebalanced across the cluster, which can impact the performance of other applications on the same network.
3. Increased storage overhead: Autoscaling can result in increased storage overhead, as additional placement groups are created to store data, which can lead to increased storage costs.
4. Complexity: Autoscaling can add complexity to Ceph's architecture, as it requires the use of additional tools and services to monitor and manage the placement groups.
5. Management: Autoscaling can be difficult to manage in Ceph, especially if there are multiple clusters with different usage patterns.

**Comment by cosminapo203** [comment]
dose anyone have any idea about this fail:
[root@controller0 ceph(keystone_admin)]# ceph -v
ceph version 14.2.22 (ca74598065096e6fcbd8433c8779a2be0c889351) nautilus (stable)
[root@controller0 ceph(keystone_admin)]# ceph mgr module enable pg_autoscaler
Error ENOENT: all mgr daemons do not support module 'pg_autoscaler', pass --force to force enablement
[root@controller0 ceph(keystone_admin)]# ceph -s
  cluster:
    id:     5037a183-fac1-4f8d-9e3a-1223914b0d47
    health: HEALTH_WARN
            mon is allowing insecure global_id reclaim

  services:
    mon: 1 daemons, quorum controller0 (age 14m)
    mgr: no daemons active
    osd: 0 osds: 0 up, 0 in

  data:
    pools:   0 pools, 0 pgs
    objects: 0 objects, 0 B
    usage:   0 B used, 0 B / 0 B avail
    pgs:

is there any guide on that

**Reply by zhucan** [reply]
@cosminapo203 please pasted the results of cmd "ceph mgr module ls"?

**Reply by zhucan** [reply]
and then run "ceph mgr module enable pg_autoscaler --force"
